src/libstd/thread/mod.rs

```diff
@@ -1052,6 +1054,28 @@ impl<T> JoinInner<T> {
 /// }).unwrap();
 /// ```
 ///
+/// Child being detached and outliving its parent:
+///
+/// ```
+/// use std::thread;
+/// use std::time::Duration;
+///
+/// let original_thread = thread::spawn(|| {
+///     let _detached_thread = thread::spawn(|| {
+///         // Here we sleep to make sure that the first thread returns before.
+///         thread::sleep(Duration::from_millis(10));
+///         // This will be called, even though the JoinHandle is dropped.
+///         println!("♫ Still alive ♫");
+///     });
+/// });
+///
+/// let _ = original_thread.join();
+/// println!("Original thread is joined.");
+/// // We make sure that the child thread has time to run, before the main
+/// // thread returns.
+/// thread::sleep(Duration::from_millis(1000));
```

Comment:
The example could indeed be sumplified to

```rust
fn main() {
    {
        thread::spawn(||{
            thread::sleep(Duration::from_millis(10));
            println!("hey");
        });
    }
    thread::sleep(Duration::from_millis(1000));
}
```

But this is not a form that I have ever encountered in real rust code, thus this example would not be very helpful.  
On the other hand a thread spawning a new thread and being joined before the new thread is a behaviour that is encountered in real rust code. It is thus a bit more complex than needed, but I believe that it is also more helpful.

There are indeed no parent-child relationship between thread, this just seemed a good illustration, if you have a better name in my I'll change for it.